.github/workflows/build.yaml

```diff
@@ -0,0 +1,34 @@
+name: Build
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+          cache: gradle
```

Comment:
I would advise removing Gradle caching, there are many secrity pitfalls with it.

Its not so bad here as there are no secrects, however in theory someone could pollute the builds from the main branch from a PR. I know its unlikely, but its not worth the risk just to save 30 seconds. (also making sure that Gradle can build from a clean slate is quite handy as well)